### DISCUSSION TITLE
Should we change the name of the default branch to main.

### DISCUSSION THREAD

**Discussion by danemorgan** [holger1411/understrap#1277] [Q&A] [answered]
The trend is toward using 'main' for the default branch name.

I propose we adopt this convention.

[edit]
As proposed by @Thomas-A-Reinert , I am adding some context here for non-US/Non-English speaking users.

Master is a fairly loaded word in the English language, and particularly in the US, where it served as the title of a slave owner. While master can mean other things, in the context of programming, and of computers, in general, it has a direct connection back to the master/slave relationship.

The word main has no such potential connotation and still offers contextual information about the status of the branch as the primary, or authoritative branch. Primary wouldn't be a bad word for this either, but typing main is faster than typing primary and seems cleaner.
[/edit]

**Comment by bacoords** [comment]
+1 to this idea. You can just pull main from master and then update the settings in the backend. I don't imagine there's anything stopping us

**Reply by danemorgan** [reply]
Yeah, the technical part of it is pretty easy, but I think we need some buy-in because, for people who are installing through Git, their locals don't automatically follow along. They have to rename locally to maintain the connection.

**Comment by Thomas-A-Reinert** [comment]
I guess the majority of non-US/non english-speaking users wouldn´t know why one should follow a "trend" that´s renaming the well-established "master" to "main". So some words about the intention behind that would be appropiate I guess.

+1 for that from my side though.

**Comment by howarddc** [comment]
+1 from me as well. Unless there is a major technical reason not to do this, I think we should move forward with it in the next round of updates.

**Comment by Thomas-A-Reinert** [comment]
Whatever the technical reasons/issues may be - Get rid of the master/slave problem. I personally don´t want Understrap to be the last repo to switch to that (now well argumented!) "trend".  I just hope/think we don´t need to put that into gender-neutral language at least.

**Comment by danemorgan** [comment, marked as answer]
Main Branch is now main.

If you have a local clone, you can update it by running:

git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
